File: packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/CozyMarkerData.m
//
//  CozyMarkerData.m
//  google_maps_flutter_ios
//
//  Created by Pietro Domingues on 24/08/23.
//

#import <Foundation/Foundation.h>
#import "CozyMarkerData.h"

@implementation CozyMarkerData
- (instancetype)initWithLabel:(NSString *)label
                      counter:(NSString *)counter
                         icon:(NSString *)icon
                   hasPointer:(BOOL)hasPointer
                   isSelected:(BOOL)isSelected
                 isVisualized:(BOOL)isVisualized
                        state:(NSString *)state
                      variant:(NSString *)variant
                         size:(NSString *)size
                   isAnimated:(BOOL)isAnimated
{
    self = [super init];
    if (self) {
        _label = label;
        _counter = counter;
        _icon = icon;
        _hasPointer = hasPointer;
        _isSelected = isSelected;
        _isVisualized = isVisualized;
        _state = state;
        _variant = variant;
        _size = size;
        _isAnimated = isAnimated;
    }
    return self;
}

- (NSString *)description {
    return [NSString stringWithFormat:@"%@ %@ %lu %@ %@ %@ %@ %@ %@ %@",
            self.label,
            self.counter,
            (unsigned long)[self.icon hash],
            self.state,
            self.variant,
            self.size,
            self.hasPointer ? @"YES" : @"NO",
            self.isSelected ? @"YES" : @"NO",
            self.isVisualized ? @"YES" : @"NO",
            self.isAnimated ? @"YES" : @"NO"
    ];
}

- (BOOL)isEqual:(CozyMarkerData *)other {
    return [[self description] isEqualToString:[other description]];
}
@end

File: packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/CozyMarkerData.java
package io.flutter.plugins.googlemaps;

public class CozyMarkerData {
    public final String label;
    public final String icon;
    public final boolean hasTail;
    public final boolean isSelected;
    public final boolean isVisualized;
    public final String state;
    public final String variant;
    public final String size;

    public CozyMarkerData(String label, String icon, boolean hasTail, boolean isSelected, boolean isVisualized, String state, String variant, String size) {
        this.label = label;
        this.icon = icon;
        this.hasTail = hasTail;
        this.isSelected = isSelected;
        this.isVisualized = isVisualized;
        this.state = state;
        this.variant = variant;
        this.size = size;
    }
    
    @Override
    public String toString() {
        return label + icon + hasTail + isSelected + isVisualized + state + variant + size;
    }
}